src/hotspot/share/opto/superword.cpp

```diff
@@ -2797,18 +2221,7 @@ bool SuperWord::is_vector_use(Node* use, int u_idx) const {
     return _packset.is_muladds2i_pack_with_pack_inputs(u_pk);
   }
 
-  if (u_pk->size() != d_pk->size()) {
-    return false;
-  }
-
-  for (uint i = 0; i < u_pk->size(); i++) {
-    Node* ui = u_pk->at(i);
-    Node* di = d_pk->at(i);
-    if (ui->in(u_idx) != di) {
-      return false;
-    }
-  }
-  return true;
+  return _packset.pack_input_at_index_or_null(u_pk, u_idx) != nullptr;
```

Comment:
Note: I introduced this method for somewhere else, and decided to refactor here as well.